python/bindings/novatel/parser.cpp

```diff
@@ -0,0 +1,54 @@
+#include "novatel_edie/decoders/oem/parser.hpp"
+
+#include "bindings_core.hpp"
+#include "json_db_singleton.hpp"
+#include "py_message_data.hpp"
+
+namespace nb = nanobind;
+using namespace nb::literals;
+using namespace novatel::edie;
+
+void init_novatel_parser(nb::module_& m)
+{
+    nb::class_<oem::Parser>(m, "Parser")
+        .def(nb::init<std::u32string>(), "json_db_path"_a)
+        .def(nb::init<JsonReader::Ptr&>(), "json_db"_a)
+        .def("__init__", [](oem::Parser* t) { new (t) oem::Parser(JsonDbSingleton::get()); }) // NOLINT(*.NewDeleteLeaks)
+        .def("load_json_db", &oem::Parser::LoadJsonDb, "json_db_path"_a)
+        .def_prop_ro("logger", &oem::Parser::GetLogger)
+        .def("enable_framer_decoder_logging", &oem::Parser::EnableFramerDecoderLogging, "level"_a = spdlog::level::debug, "filename"_a = "edie.log")
+        .def_prop_rw("ignore_abbreviated_ascii_responses", &oem::Parser::GetIgnoreAbbreviatedAsciiResponses,
+                     &oem::Parser::SetIgnoreAbbreviatedAsciiResponses)
+        .def_prop_rw("decompress_range_cmp", &oem::Parser::GetDecompressRangeCmp, &oem::Parser::SetDecompressRangeCmp)
+        .def_prop_rw("return_unknown_bytes", &oem::Parser::GetReturnUnknownBytes, &oem::Parser::SetReturnUnknownBytes)
+        .def_prop_rw("encode_format", &oem::Parser::GetEncodeFormat, &oem::Parser::SetEncodeFormat)
+        .def_prop_rw("filter", &oem::Parser::GetFilter, &oem::Parser::SetFilter)
+        .def("write",
+             [](oem::Parser& self, const nb::bytes& data) { return self.Write(reinterpret_cast<const uint8_t*>(data.c_str()), data.size()); })
+        .def("read",
```

Comment:
Thinking more about this comment I wonder if, in the future, it would be better to separate conversion (binary -> ascii) from programmatic access. The current API is a bit confusing where if you don't convert to flattened_binary the message variable is a None. Future thing to think about.